src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> currentNode = table[getKeyIndex(key)];
+        if (currentNode == null) {
+            table[getKeyIndex(key)] = new Node<>(key, value, null);
+        } else {
+            while (currentNode.next != null
+                    || Objects.equals(key, currentNode.key)) {
+                if (Objects.equals(key, currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
```

Comment:
this line can produce NullPointerException. You would better check in while(currentNode != null) and then you will not need  line 27. If above while remove at all and move line 24 after while. Also in while you have to add one check if(currentNode.next == null) {
insert new node here
}

Reply:
done